### PR TITLE
server/api: add admin operator

### PR DESCRIPTION
**Pull request by huachaohuang**
Support API to add operator to transfer leader or region to some stores.

### CODE REVIEW

**Comment by huachaohuang** [comment]
PTAL @siddontang @disksing

**Comment by disksing** [comment]
LGTM, PTAL @siddontang @overvenus

File: server/coordinator.go [hunk 1]
```diff
@@ -181,6 +181,13 @@ func (c *coordinator) addOperator(op Operator) bool {
 	defer c.Unlock()
 
 	regionID := op.GetRegionID()
+
+	// Admin operator bypasses the check.
+	if op.GetResourceKind() == adminKind {
+		c.operators[regionID] = op
```
Comment (siddontang) [inline server/coordinator.go]: no need to close old operator if exists?

Comment (huachaohuang) [inline server/coordinator.go]: Operator has no "close", we just replace it with an admin operator.

File: server/api/operator.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	switch name {
+	case "transfer-leader":
+		regionID, ok := input["region_id"].(float64)
+		if !ok {
+			h.r.JSON(w, http.StatusBadRequest, "missing region id")
+			return
+		}
+		storeID, ok := input["leader_store_id"].(float64)
```
Comment (siddontang) [inline server/api/operator.go]: s/leader_store_id/to_store_id/

File: server/api/operator.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+			h.r.JSON(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+	case "transfer-region":
+		regionID, ok := input["region_id"].(float64)
+		if !ok {
+			h.r.JSON(w, http.StatusBadRequest, "missing region id")
+			return
+		}
+		storeIDs, ok := parseStoreIDs(input["region_store_ids"])
```
Comment (siddontang) [inline server/api/operator.go]: I prefer transfer region one by one.

Comment (huachaohuang) [inline server/api/operator.go]: How about adding a "transfer-peer" operator to transfer only one peer?

Comment (siddontang) [inline server/api/operator.go]: +1

File: server/api/operator.go [hunk 4]
```diff
@@ -122,19 +122,42 @@ func (h *operatorHandler) Post(w http.ResponseWriter, r *http.Request) {
...
 		if len(storeIDs) == 0 {
-			h.r.JSON(w, http.StatusBadRequest, "missing region store ids")
+			h.r.JSON(w, http.StatusBadRequest, "missing store ids to transfer region to")
 			return
 		}
 		if err := h.AddTransferRegionOperator(uint64(regionID), storeIDs); err != nil {
 			h.r.JSON(w, http.StatusInternalServerError, err.Error())
 			return
 		}
+	case "transfer-peer":
```
Comment (siddontang) [inline server/api/operator.go]: do we need to convert `-` to `_` ?

Comment (huachaohuang) [inline server/api/operator.go]: I prefer using `-` in string, like our schedulers' names.

**Pull request merged**